### DISCUSSION TITLE
Adding Contabo S3 like API to Directus

### DISCUSSION THREAD

**Discussion by loulou2852** [directus/directus#13863] [Feature Requests]
Hello,

Recently, Contabo launches Object storage, but as it's written in their docs:
"Virtual Hosted Buckets are not supported. Path based buckets are being used. so e.g. eu2.contabostorage.com/bucketname is used instead of bucketname.eu2.contabostorage.com".

I tried to connect with the S3 API but due to lack of bucket's subdomain, it's not working for me, or maybe I didn't put it right?

Any possibility to add it into the providers?

I've got an account with Contabo storage, so I can give you an access if necessary for testing purposes

Regards.

Louis

**Comment by azrikahar** [comment]
> Path based buckets are being used. so e.g. eu2.contabostorage.com/bucketname is used instead of bucketname.eu2.contabostorage.com

You can configure the [`s3ForcePathStyle` option](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#:~:text=s3ForcePathStyle%20(Boolean)%20%E2%80%94%20whether%20to%20force%20path%20style%20URLs%20for%20S3%20objects.) of the underlying AWS sdk s3 client via `STORAGE_<LOCATION>_S3_FORCE_PATH_STYLE=true`, which should make it work if I understood the problem correctly. Also see https://github.com/directus/directus/issues/10392#issuecomment-988895670 for more info.

**Reply by loulou2852** [reply]
Thank you, for your reply. 

I didn't know what this Env var meant. It's working now.

Thank you :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
